src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala

```diff
@@ -450,7 +450,18 @@ class SchedulerActions(
             Done
         }.foreach { _ =>
           logger.info(s"Killing instances ${instances.map(_.instanceId)}")
-          killService.killInstances(instances, KillReason.OverCapacity)
+
+          def killInstances(instances: Seq[Instance]): Future[Done] = async {
+            val changeGoalsFuture = instances.map { i =>
+              if (i.hasReservation) instanceTracker.goalStopped(i.instanceId)
+              else instanceTracker.goalDecommissioned(i.instanceId)
+            }
+
+            await(Future.sequence(changeGoalsFuture))
```

Comment:
Marathon is full of this patter - but what happens if one the instance state transitions fail e.g. because of a timeout in *InstanceTrackerDelegate*? Some of the instances will the the new `Goal` state, some wouldn't and the subsequent `killService.killInstances` will never be called.

Reply:
hmm that's a fair point... would you retry in that case?

It actually does not make much difference now but LATER Scheduler will use the goal to actually issue the kill so then it would be super-important to actually have the goals set correctly... also then the following line with kill would be removed

Reply:
> hmm that's a fair point... would you retry in that case?

Ideally, we **would not** retry ourselves - retry should be either user re-triggered (because the user got an exception in the API) or Mesos re-triggered (because we didn't ACK-ed the corresponding status update). But I fear that in this case, this call will end up in limbo.